scripts/generate_icon_types.js

```diff
@@ -0,0 +1,234 @@
+// Script to generate TypeScript icon type definitions from filesystem
+import { existsSync, readdirSync, statSync, writeFileSync, mkdirSync } from 'fs';
+import { resolve, join, dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+// ESM equivalent of __dirname
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+// Paths
+const iconsOutlinePath = resolve(__dirname, '../assets/icons/outline');
+const iconsSolidPath = resolve(__dirname, '../assets/icons/solid');
+const buildPath = resolve(__dirname, '../build');
+const outputJsPath = join(buildPath, 'icons.js');
+const outputDtsPath = join(buildPath, 'icons.d.ts');
+
+/**
+ * Recursively get all SVG files from a directory
+ * @param {string} dirPath - Directory to scan
+ * @returns {string[]} - Array of icon names (without .svg extension)
+ */
+function getIconNames(dirPath) {
+  const iconNames = [];
+
+  function traverse(currentPath) {
+    if (!existsSync(currentPath)) {
+      return;
+    }
+
+    const items = readdirSync(currentPath);
```

Comment:
aqui nesse ponto se você considerar fazer essa alteração:


```suggestion
    const items = readdirSync(currentPath, { withFileTypes: true });
```

será retornado objetos que já contém a informação se é um diretório ou arquivo
fonte: https://www.geeksforgeeks.org/node-js/node-js-fs-readdirsync-method/